commit 2e751dfb5ffd20d1a31837dbc9718741df69bffe
Merge: a0e136d436de 7b6b46311a85
Author: Paolo Bonzini <pbonzini@redhat.com>
Date:   Thu Feb 9 16:01:23 2017 +0100

    Merge tag 'kvmarm-for-4.11' of git://git.kernel.org/pub/scm/linux/kernel/git/kvmarm/kvmarm into HEAD
    
    kvmarm updates for 4.11
    
    - GICv3 save restore
    - Cache flushing fixes
    - MSI injection fix for GICv3 ITS
    - Physical timer emulation support

diff --cc arch/powerpc/mm/init-common.c
index 2be5dc242832,f2108c40e697..eb8c6c8c4851
--- a/arch/powerpc/mm/init-common.c
+++ b/arch/powerpc/mm/init-common.c
@@@ -79,11 -78,15 +79,15 @@@ void pgtable_cache_add(unsigned shift, 
  	align = max_t(unsigned long, align, minalign);
  	name = kasprintf(GFP_KERNEL, "pgtable-2^%d", shift);
  	new = kmem_cache_create(name, table_size, align, 0, ctor);
+ 	if (!new)
+ 		panic("Could not allocate pgtable cache for order %d", shift);
+ 
  	kfree(name);
  	pgtable_cache[shift - 1] = new;
+ 
  	pr_debug("Allocated pgtable cache for order %d\n", shift);
  }
 -
 +EXPORT_SYMBOL_GPL(pgtable_cache_add);	/* used by kvm_hv module */
  
  void pgtable_cache_init(void)
  {